container/go/pkg/compat/overrideConfig.go

```diff
@@ -0,0 +1,453 @@
+	}
+	formatArgs := make(map[string]interface{})
+	for _, infofile := range stampInfoFile {
+		f, err := os.Open(infofile)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to open file %s", infofile)
+		}
+		defer f.Close()
+		// scanner reads line by line and discards '\n' character by default.
+		scanner := bufio.NewScanner(f)
+		var temp []string
+		for scanner.Scan() {
+			temp = strings.Split(scanner.Text(), " ")
+			key, val := temp[0], temp[1]
+			if _, ok := formatArgs[key]; ok {
+				fmt.Printf("WARNING: Duplicate value for key %s: using %s", key, val)
+			}
+			formatArgs[key] = val
+		}
+		if err = scanner.Err(); err != nil {
+			return "", errors.Wrapf(err, "failed to read line from file %s", infofile)
+		}
+	}
+	// do string manipulation in order to mimic python string format.
+	// specifically, replace '{' with '{{.' and '}' with '}}'.
+	inpReformatted := strings.ReplaceAll(inp, "{", "{{.")
+	inpReformatted = strings.ReplaceAll(inpReformatted, "}", "}}")
+	return formatWithMap(inpReformatted, formattedString(formatArgs)), nil
+}
+
+// mapToKeyValue reverses a map to a '='-separated array of strings in {key}={value} format.
+func mapToKeyValue(kvMap map[string]string) []string {
+	keyVals := []string{}
+	concatenated := ""
+	for k, v := range kvMap {
+		concatenated = k + "=" + v
+		keyVals = append(keyVals, concatenated)
+	}
+
+	sort.Sort(sort.StringSlice(keyVals))
+	return keyVals
+}
+
+// resolveVariables resolves the environment variables embedded in the given value using
+// provided map of environment variable expansions.
+// It handles formats like "PATH=$PATH:..."
+func resolveVariables(value string, environment map[string]string) (string, error) {
+	fmt.Printf("environment override %v", environment)
+	var i interface{}
+	i = "this is a string type"
+	var errorMet = false
+	mapper := func(p string) string {
+		switch i.(type) {
+		case string:
+			return environment[p]
+		}
+		errorMet = true
+		return ""
+	}
+	if errorMet {
+		err := errors.New("environment variable sought after does not exist in environement map")
+		if err != nil {
+			errors.Wrap(err, "failed to create new error")
+		}
+		return "", err
+	}
+	return os.Expand(value, mapper), nil
+}
+
+// OverrideImageConfig updates the current image config file to reflect the given changes.
+func OverrideImageConfig(overrideInfo *OverrideConfigOpts) error {
+	overrideInfo.ConfigFile.Author = "Bazel"
+	overrideInfo.ConfigFile.OS = overrideInfo.OperatingSystem
+	overrideInfo.ConfigFile.Architecture = defaultProcArch
+
+	var err error
+	// createTime stores the input creation time with macros substituted.
+	var createTime string
+	// creationTime is the RFC 3339 formatted time derived from createTime input.
+	creationTime, err := time.Parse(time.RFC3339, defaultTimestamp)
+	log.Printf("the very very first creationTime: %v", creationTime)
+	if err != nil {
+		return errors.Wrap(err, "Unable to parse the default unix epoch time 1970-01-01T00:00:00Z")
```

Comment:
messages in error objects begin with small letters. Also, instead of harcoding the time here, just use `defaultTimestamp`